payments/schema/mutations.py

```diff
@@ -396,52 +396,66 @@ def mutate_and_get_payload(cls, root, info, **input):
-                lease = get_node_from_global_id(
-                    info, lease_id, WinterStorageLeaseNode, nullable=True
-                )
-            finally:
-                if not lease:
-                    raise VenepaikkaGraphQLError(
-                        _("Lease with the given ID does not exist")
-                    )
-                input["lease"] = lease
+    def mutate_and_get_payload(cls, root, info, orders, **input):
 
-        try:
-            # handle case where input has both lease and status.
-            # set order status only after changing the lease, because setting order status
-            # usually triggers a change in lease status.
-            new_status = input.pop("status", None)
-            update_object(order, input)
-            if new_status:
-                order.set_status(new_status, _("Manually updated by admin"))
+        successful_orders = []
+        failed_orders = []
 
-        except (ValidationError, IntegrityError) as e:
-            raise VenepaikkaGraphQLError(e)
-        return UpdateOrderMutation(order=order)
+        for order_input in orders:
+            order_id = order_input.pop("id")
+            try:
+                with transaction.atomic():
+                    order = get_node_from_global_id(
+                        info, order_id, only_type=OrderNode, nullable=False
+                    )
+                    if lease_id := order_input.pop("lease_id", None):
+                        lease = None
+                        try:
+                            lease = get_node_from_global_id(
+                                info, lease_id, BerthLeaseNode, nullable=True
+                            )
+                        # If a different node type is received get_node raises an assertion error
+                        # when trying to validate the type
+                        except AssertionError:
+                            lease = get_node_from_global_id(
+                                info, lease_id, WinterStorageLeaseNode, nullable=True
+                            )
+                        finally:
+                            if not lease:
+                                raise VenepaikkaGraphQLError(
+                                    _("Lease with the given ID does not exist")
+                                )
+                            order_input["lease"] = lease
+
+                    # handle case where order_input has both lease and status.
+                    # set order status only after changing the lease, because setting order status
+                    # usually triggers a change in lease status.
+                    new_status = order_input.pop("status", None)
+                    update_object(order, order_input)
+                    if new_status:
+                        order.set_status(new_status, _("Manually updated by admin"))
+
+            except (ValidationError, IntegrityError, VenepaikkaGraphQLError) as e:
```

Comment:
We should also catch here for `OrderStatusTransitionError`. If you try to move one order to an invalid state, the whole request will fail because the exception is not caught.

Reply:
good catch, fixed

Reply:
Great! Now it did catch the error

```json
{
  "data": {
    "updateOrders": {
      "failedOrders": [
        {
          "id": "T3JkZXJOb2RlOjE2NmNlZTdiLWYyMDYtNDZmYy05NjVmLWU1NmUwZjM0NmIwMg==",
          "error": "Cannot set order co73zgdhkwcqa state to \"refunded\", it is in an invalid state \"drafted\"."
        }
      ],
      "successfulOrders": [
        {
          "id": "T3JkZXJOb2RlOjUwNGQ5NzQ4LWUyMDQtNGZlNy05OWFiLTQ5NWJmZDBlZDcxNg==",
          "comment": "Another new comment!",
          "price": "100.00",
          "taxPercentage": "24.00",
          "dueDate": "2021-11-30",
          "status": "DRAFTED",
          "customer": {
            "id": "UHJvZmlsZU5vZGU6YzRhYzljNjUtYzdjYy00OTkyLWFkYzEtNjZkMGM0YmRlZjA3"
          },
          "product": null,
          "lease": null
        }
      ]
    }
  }
}
```